src/Models/Context/OatPageContext/OatPageContextUtils.ts

```diff
@@ -402,76 +402,46 @@ export const getTotalReferenceCount = (models: DtdlInterface[]): number => {
+    models: DtdlInterface[]
+): {
+    relationshipCount: number;
+    inheritanceCount: number;
+    componentCount: number;
+    v2ModelCount: number;
+    v3ModelCount: number;
+} => {
+    let relationshipCount = 0;
     let inheritanceCount = 0;
-    models.forEach((m) => {
-        if (!m.extends) {
-            return;
-        } else if (typeof m.extends == 'object') {
-            inheritanceCount = inheritanceCount + m.extends.length;
-        } else {
-            inheritanceCount = inheritanceCount + 1;
-        }
-    });
-    return inheritanceCount;
-};
-
-/** get property count */
-export const getTotalPropertyCount = (models: DtdlInterface[]): number => {
-    const countProperties = (contents: DtdlInterfaceContent[]): number => {
-        let count = 0;
-        contents.forEach((c) => {
-            if (isDTDLProperty(c)) count = count + 1;
-        });
-        return count;
-    };
-
-    let propertyCount = 0;
-    models.forEach((m) => {
-        if (!m.contents) {
-            return;
-        } else {
-            propertyCount = propertyCount + countProperties(m.contents);
-        }
-    });
-    return propertyCount;
-};
-
-/** get component count */
-export const getTotalComponentCount = (models: DtdlInterface[]): number => {
-    const countComponents = (contents: DtdlInterfaceContent[]): number => {
-        let count = 0;
-        contents.forEach((c) => {
-            if (isDTDLComponentReference(c)) count = count + 1;
-        });
-        return count;
-    };
-
     let componentCount = 0;
-    models.forEach((m) => {
-        if (!m.contents) {
-            return;
-        } else {
-            componentCount = componentCount + countComponents(m.contents);
-        }
-    });
-    return componentCount;
-};
-
-/** get model version count */
-export const getModelVersionCount = (
-    models: DtdlInterface[]
-): [number, number] => {
     let v2ModelCount = 0;
     let v3ModelCount = 0;
+
     models.forEach((m) => {
-        const version = getDtdlVersionFromContext(m['@context']);
-        version === '2'
-            ? (v2ModelCount = v2ModelCount + 1)
-            : (v3ModelCount = v3ModelCount + 1);
+        // Calculate relationships
+        const relationships =
+            m.contents.filter((x) => isDTDLRelationshipReference(x)) ?? [];
+        relationshipCount += relationships.length;
+        // Calculate inheritance
+        inheritanceCount += ensureIsArray(m.extends).length;
+        // Calculate components
+        const components =
+            m.contents.filter((x) => isDTDLComponentReference(x)) ?? [];
+        componentCount += components.length;
+        // Calculate model versions
+        modelHasVersion3Context(m) ? (v3ModelCount += 1) : (v2ModelCount += 1);
     });
-    return [v2ModelCount, v3ModelCount];
+
+    return {
+        relationshipCount: relationshipCount,
+        inheritanceCount: inheritanceCount,
+        componentCount: componentCount,
```

Comment:
is the goal really to know the total number of component references in an ontology? or to know the number of references on a given node and then we can average it per model in an ontology? I'm just wondering what these counts will tell us. 🤔

Reply:
What do you think would be the most helpful information? I just have context on how to do telemetry stuff, you have better insight into what each thing is.

Reply:
It really depends what we are trying to learn. Maybe Riley just wants to know if people ever use Components or something, this would do that for us. Let's send it.